test/streams/go2go.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+chai.use(require('dirty-chai'))
+chai.use(require('chai-bytes'))
+const expect = chai.expect
+const { decode } = require('length-prefixed-stream')
+const { pipeline } = require('readable-stream')
+const { StreamInfo } = require('libp2p-daemon/src/protocol')
+const { ends } = require('libp2p-daemon/src/util')
+
+const spawnDaemons = require('../utils/spawnDaemons')
+
+describe('streams', () => {
+  let daemons
+
+  // Start daemons
+  before(async function () {
+    this.timeout(20 * 1000)
+
+    daemons = await spawnDaemons(2, 'go')
+
+    // connect them
+    const identify0 = await daemons[0].client.identify()
+    await daemons[1].client.connect(identify0.peerId, identify0.addrs)
+  })
+
+  // Stop daemons
+  after(async function () {
+    await Promise.all(
+      daemons.map((daemon) => daemon.stop())
+    )
+  })
+
+  it('go stream to go listener', async function () {
+    this.timeout(10 * 1000)
+
+    const sendingPeerId = await daemons[0].client.identify().peerId
+    const { receivingPeerId, receivingAddrs } = await daemons[1].client.identify()
+
+    await daemons[1].client.startServer(receivingAddrs[0], async (conn) => {
+      // Decode the stream
+      const dec = decode()
+      conn.pipe(dec)
+
+      // Read the stream info from the daemon, then pipe to echo
+      const message = await ends(dec).first()
+      let response = StreamInfo.decode(message)
+
+      expect(response.peer).to.eql(sendingPeerId.toBytes())
+      expect(response.proto).to.eql('/echo/1.0.0')
+
+      conn.unpipe(dec)
+
+      // Echo messages
+      pipeline(
```

Comment:
I think we do not need to echo messages, being able to send and receive, as done in the `js-libp2p-dameon-client` should be enough

Reply:
Removing